Set up project settings, and set up initial tableView in storyboard

jmgoldb opened the pull request:
Also did best to clean up boilerplate code.

jmgoldb committed:
Set up project settings, and set up initial tableView in storyboard

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,33 @@
+
+@implementation BoxOfficeViewController
+
+#pragma mark - Table view data source
```
I know this came from Xcode with the table view controller template, but if you change this to `#pragma mark - UITableViewDataSource methods`, then you can option- or command-click on `UITableViewDataSource` to see the mini-docs or jump to the definition, respectively.

jmgoldb committed:
Added initial mock data to populate initial tableView with two BoxOfficeCells upon startup.

jmgoldb committed:
Added image loading from mock data.  Also successfully implemented requesting an ordered list of the top 10 box office movies via RottenTomatoes' api

jmgoldb committed:
Implemented pull request comment changes, and implemented loading of images for movies.  Need to hunt down memory leak.

jmgoldb commented:
Adressed comments on pragma mark, and added in loading of movie titles and images

jmgoldb committed:
Uncommented line of coded needed for program to compile.  Oops!

jmgoldb committed:
Image Loading works as intended

jmgoldb committed:
Implemented rudimentary movie details.  Removed sales data from box office listings, as that data is not provided by the Rotten Tomatoes API.

jmgoldb committed:
Detail View now has dynamic title

vokal-isaac commented on MovieChecker/MovieChecker.xcodeproj/project.pbxproj:
```diff
@@ -7,13 +7,20 @@
+		E46B8A301A715A9E001372EE /* Movie.m in Sources */ = {isa = PBXBuildFile; fileRef = E46B8A2F1A715A9E001372EE /* Movie.m */; };
+		E46B8A331A715F51001372EE /* MovieDetailViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = E46B8A321A715F51001372EE /* MovieDetailViewController.m */; };
+		E46B8A371A7162D7001372EE /* bomb.jpg in Resources */ = {isa = PBXBuildFile; fileRef = E46B8A351A7162D7001372EE /* bomb.jpg */; };
+		E46B8A381A7162D7001372EE /* hit.jpg in Resources */ = {isa = PBXBuildFile; fileRef = E46B8A361A7162D7001372EE /* hit.jpg */; };
```
Do you still need these two images?

vokal-isaac commented on MovieChecker/MovieChecker/AppDelegate.m:
```diff
@@ -7,39 +7,28 @@
-    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+{
+    NSMutableArray *_movies;
 }
```
Generally, you shouldn't need to explicitly declare ivars.  In particular, I'm not sure what this ivar is being used for.

vokal-isaac commented on MovieChecker/MovieChecker/AppDelegate.m:
```diff
@@ -7,39 +7,28 @@
+    
+    UINavigationController *navigationController = (UINavigationController *)self.window.rootViewController;
+    BoxOfficeViewController *boxOfficeViewController = navigationController.viewControllers.firstObject;
+    boxOfficeViewController.movies = _movies;
```
Why does this happen here?  Could the `movies` property of the `BoxOfficeViewController` be set to an empty mutable array in the initializer or `viewDidLoad` of `BoxOfficeViewController` instead?

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCell.h:
```diff
@@ -0,0 +1,17 @@
+
+@property (nonatomic, weak) IBOutlet UILabel *nameLabel;
+@property (nonatomic, weak) IBOutlet UILabel *salesLabel;
+@property (nonatomic, weak) IBOutlet UIImageView *thumbnailImageView;
```
This is fine for a first iteration; generally, it's preferable to have the cell subclass expose a method like `updateWithSomeModel:(SomeModel *)someModel`, from which the actual UI elements are set, and keep these `IBOutlet`s private.

jmgoldb replied:
Changed these here, but in BoxOfficeCollectionViewCell, it seems that I need to expose thumbnailImageView, in order to work with SDWebImage, unless I want to move the image downloading method calls to the Cell's implementation, which I probably don't.  (I might also be able to make a dummy UIImageView, and set the cell's image to match, but I don't know if that's a good idea.)

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCell.m:
```diff
@@ -0,0 +1,25 @@
+- (void)awakeFromNib
+{
+    // Initialization code
+}
```
1. On iOS, you should always call `[super awakeFromNib]` in any implementation of `awakeFromNib`.
2. Unless there's a reason to keep it, delete empty methods; if there's a reason to keep it, document it with a comment.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCell.m:
```diff
@@ -0,0 +1,25 @@
+    [super setSelected:selected animated:animated];
+
+    // Configure the view for the selected state
+}
```
Essentially empty method.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.h:
```diff
@@ -8,8 +8,8 @@
-@interface ViewController : UIViewController
+@interface BoxOfficeViewController : UITableViewController
 
+@property (nonatomic, strong) NSMutableArray *movies;
```
Conceptually, why expose this property?  Do any other classes need to know about it?  If so, why?

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+#import "BoxOfficeCell.h"
+#import "Movie.h"
+#import "RottenTomatoes.h"
+#import "MovieDetailViewController.h"
```
Organize `#import` statements according to the coding standards.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    if (self) {
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+        //[config setHTTPAdditionalHeaders:@{@"apikey": [RottenTomatoes apiKey]}];
+        self.session = [NSURLSession sessionWithConfiguration:config];
```
It's preferable to directly set the ivar in an initializer method, rather than calling the setter (which is what happens with dot notation), but in this case, I'd think it might make more sense to move this setup to `viewDidLoad` and remove the initializer entirely.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    [self moviesOnRottenTomatoes];
+}
+
+- (void)moviesOnRottenTomatoes
```
I'd expect a method named `moviesOnRottenTomatoes` to return an array of movies, which this doesn't.  Perhaps `loadMoviesFromRottenTomatoes`?

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+            }
+        } else {
+            // TODO: Handle the error!
+        }
```
Consider rewriting this as:

``` objc
        if (error) {
            // TODO: Handle the error!
            return;
        }
        NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
        if (httpResp.statusCode != 200) {
            // TODO: Handle bad status code!
            return;
        }
        [self interpretBoxOfficeMoviesFromArray:[RottenTomatoes arrayFromData:data withKey:@"movies"]];
```

The basic idea is to test for failure/error conditions and handle them (and end execution with a `return`) in the `if`-blocks, while the mainline of execution (at the primary indent level) is the expected/successful flow.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+        movie.year = [movieData[@"year"] intValue];
+        movie.runtime = [movieData[@"runtime"] intValue];
+        [self.movies addObject:movie];
+    }
```
Converting the raw data into model objects probably doesn't belong in the view controller; perhaps in the model or in some intermediate/helper controller class.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
```
This method is not needed if you're just returning `1`.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    // TODO: Fix Image Loading to not be laggy.
```
Is this `TODO` still relevant?

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    // TODO: Fix Image Loading to not be laggy.
+    BoxOfficeCell *boxOfficeCell = (BoxOfficeCell *)[tableView dequeueReusableCellWithIdentifier:@"BoxOfficeMovieCell" forIndexPath:indexPath];
```
It's best to avoid inline string literals for things like this (search, e.g., "stringly typed code").  If you make your identifier the same as the cell class name, you can use `NSStringFromClass()` to get the identifier without having the string literal.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+                                                               //DLOG(@"");
+                                                               movie.image = downloadedImage;
+                                                               dispatch_async(dispatch_get_main_queue(), ^{
+                                                                   if (cell.tag == indexPath.row) {
```
It's probably worth a comment here explaining what this is doing.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+                                                               UIImage *downloadedImage = [[UIImage alloc] initWithData:data];
+                                                               //DLOG(@"");
+                                                               movie.image = downloadedImage;
+                                                               dispatch_async(dispatch_get_main_queue(), ^{
```
It's slightly preferable to use `[[NSOperationQueue mainQueue] addOperationWithBlock:^{ ... }];`, since it's a higher-level/more-idiomatically-Objective-C construct than the lower-level/more-C-like `dispatch_async()` call (the two are basically equivalent).

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    NSURL *url = [NSURL URLWithString:movie.imagePath];
+    NSURLSessionDataTask *downloadImageTask = [[NSURLSession sharedSession]
+                                                   dataTaskWithURL:url
+                                                   completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
```
There's something funky with the indentation here.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    Movie *movie = self.movies[indexPath.row];
+    boxOfficeCell.nameLabel.text = movie.name;
+    boxOfficeCell.thumbnailImageView.image = nil;
+    [self imageForMovie:movie forCell:boxOfficeCell atIndexPath:indexPath];
```
3 or more parameters => one parameter per line.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    boxOfficeCell.thumbnailImageView.image = nil;
+    [self imageForMovie:movie forCell:boxOfficeCell atIndexPath:indexPath];
+        
+    
```
Extra blank line.

vokal-isaac commented on MovieChecker/MovieChecker/Info.plist:
```diff
@@ -22,8 +22,6 @@
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
-	<key>UILaunchStoryboardName</key>
-	<string>LaunchScreen</string>
```
If you're not using the LaunchScreen xib, you can probably delete it from the project.

vokal-isaac commented on MovieChecker/MovieChecker/Movie.h:
```diff
@@ -0,0 +1,26 @@
+@property (nonatomic, strong) NSString *name;
+@property (nonatomic, strong) NSString *imagePath;
+@property (nonatomic, strong) NSString *urlPath;
+@property (nonatomic, assign) BOOL imageLoaded;
```
Is this being used anymore?

vokal-isaac commented on MovieChecker/MovieChecker/Movie.m:
```diff
@@ -0,0 +1,27 @@
+        self.imagePath = imagePath;
+        self.urlPath = urlPath;
+    } else {
+        // TODO: Handle Error?
```
Nope.  If `[super init]` returns `nil`, you should just let this method return `nil`, too.

vokal-isaac commented on MovieChecker/MovieChecker/Movie.h:
```diff
@@ -0,0 +1,26 @@
+@property (nonatomic, assign) NSInteger runtime;
+@property (nonatomic, strong) NSString *synopsis;
+
+- (Movie *)initWithTitle:(NSString *)name imagePath:(NSString *)imagePath urlPath:(NSString *)urlPath;
```
3 or more parameters => one parameter per line.

vokal-isaac commented on MovieChecker/MovieChecker/Movie.m:
```diff
@@ -0,0 +1,27 @@
+
+@implementation Movie
+
+- (Movie *)initWithTitle:(NSString *)name imagePath:(NSString *)imagePath urlPath:(NSString *)urlPath;
```
3 or more parameters => one parameter per line.

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.h:
```diff
@@ -0,0 +1,21 @@
+@property (nonatomic, weak) IBOutlet UILabel *movieTitle;
+@property (nonatomic, weak) IBOutlet UILabel *rating;
+@property (weak, nonatomic) IBOutlet UILabel *synopsis;
+@property (weak, nonatomic) IBOutlet UILabel *year;
```
The order of `atomic`/`nonatomic` and `weak`/`strong`/`assign` isn't specified in the standards, but you should be consistent in the way you order them, at least on a per-file basis of not a per-project basis.

vokal-isaac replied:
Also, do these `IBOutlet`s need to be publicly exposed?

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.m:
```diff
@@ -0,0 +1,52 @@
+    [super viewDidLoad];
+    self.title = self.movie.name;
+    self.movieTitle.text = self.movie.name;
+    self.year.text = [NSString stringWithFormat:@"%ld", (long)self.movie.year];
```
It's slightly preferable to use `%@` and `@`-box the value rather than cast it; i.e., `[NSString stringWithFormat:@"%@", @(self.movie.year)]`... though since you just want to turn the `NSInteger` into a string, you could call `description` on the `@`-boxed value: `[@(self.movie.year) description]`

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.m:
```diff
@@ -0,0 +1,52 @@
+        default:
+            return 0; // Error
+    }
+}
```
If you're using a static table view in a storyboard, I don't think you need `numberOfSectionsInTableView:` or `tableView:numberOfRowsInSection:`.

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.m:
```diff
@@ -0,0 +1,52 @@
+    switch (section) {
+        case 0:
+            return 3;
+            break;
```
Why `break` here and not in the other cases?

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoes.h:
```diff
@@ -0,0 +1,18 @@
++ (NSURL *)appRootURL;
++ (NSString *)apiKey;
++ (NSURL *)boxOfficeURL;
++ (NSArray *)arrayFromData:(NSData *)data withKey:(NSString *)key;
```
Why pass in the key?  Would it make sense for the key to be a constant within this class's implementation?

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoes.m:
```diff
@@ -0,0 +1,50 @@
+static NSString *apiKey = @"jw64knjg769gnmd2zs35ttz6";
+static NSString *homeUrlRoot = @"http://api.rottentomatoes.com/api/public/v1.0.json?apikey=";
+static NSString *baseURL = @"http://api.rottentomatoes.com/api/public/v1.0/";
+static NSString *boxOfficeURL = @"lists/movies/box_office.json?limit=50";
```
In our standards, `static` variable names should begin with a capital letter.

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoes.m:
```diff
@@ -0,0 +1,50 @@
++ (NSURL *)boxOfficeURL
+{
+    // TODO: Does this need to be cleaned up?
+    NSString *url = [[NSString stringWithString:baseURL] stringByAppendingFormat:@"%@&apikey=%@", boxOfficeURL,apiKey];
```
Space after the comma.

vokal-isaac replied:
Also, it's probably better to use something like `URLWithString:relativeToURL:` rather than string-concatenating URLs.

jmgoldb replied:
URLWithString:relativeToURL: seems to only work for paths, and does not append parameters correctly, so leaving this as-is (with the space after the comma added, of course).

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeViewController.m:
```diff
@@ -0,0 +1,148 @@
+    self = [super initWithCoder:aDecoder];
+    if (self) {
+        NSURLSessionConfiguration *config = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+        //[config setHTTPAdditionalHeaders:@{@"apikey": [RottenTomatoes apiKey]}];
```
Is this still needed?

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoes.h:
```diff
@@ -0,0 +1,18 @@
+@interface RottenTomatoes : NSObject
+
++ (NSURL *)appRootURL;
++ (NSString *)apiKey;
```
Does this need to be exposed outside the class?

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoes.m:
```diff
@@ -0,0 +1,50 @@
+    NSDictionary *jsonData = [NSJSONSerialization JSONObjectWithData:data
+                                                             options:NSJSONReadingAllowFragments
+                                                               error:&jsonError];
+    if (!jsonError) {
```
For methods that take an `NSError **` parameter, it's incorrect to use the error to test for success.  You have to test the return value of the method—if it's `nil` or `NO` (basically, false-y), then there's an error and you can look at the error parameter; if it's non-false-y, then the error parameter may or may not still be `nil`.

jmgoldb committed:
Added pods, and switched from tableviews to collections

jmgoldb committed:
Implemented most (but not all) of changes based on feedback from Jan 26th code review.  Next up is self-review of existing code written after that code review

jmgoldb committed:
Removed ObjectiveSugar, finished addressing code review comments, and tidied up new code.

jmgoldb commented:
Requesting a new code review, please!

jmgoldb committed:
Gutted old, obsolete classes and items in the storyboard.  (Might rename BoxOfficeCell to BoxOfficeTableCell, and BoxOfficeCollectionViewCell to just BoxOfficeCollectionCell).

Also added in support for alternate layout attributes using different cells.  Alternate layout needs to be implemented next.

jmgoldb committed:
Changed names of Cell classes

jmgoldb committed:
Implemented testing/Mock Data for retrieval of box office data from Rotten Tomatoes' API.  Also reorganized all #imports to conform to Vokal Coding Standards

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCollectionCell.h:
```diff
@@ -0,0 +1,15 @@
+
+@interface BoxOfficeCollectionCell : UICollectionViewCell
+
+- (void)setThumbnailImageFromURL:(NSURL *)url;
```
It would probably be preferable to have something like a `displayMovie:` method that takes a model object and leaves the details of what exactly is being done with that movie object up to the cell.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCollectionCell.m:
```diff
@@ -0,0 +1,31 @@
+                                                                NSError *error,
+                                                                SDImageCacheType cacheType,
+                                                                NSURL *imageURL) {
+        self.thumbnailImageView.image = image;
```
It would probably be safer to use a weak reference to `self` rather than letting `self` be strongly captured by the completion block.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCollectionViewController.m:
```diff
@@ -0,0 +1,172 @@
+        NSIndexPath *path = [self.collectionView indexPathForCell:sender];
+        Movie *movie = self.movies[path.row];
+        MovieDetailViewController *detailViewController = segue.destinationViewController;
+        [detailViewController setMovie:movie];
```
If `movie` is a property of `detailViewController`, then use dot notation.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeCollectionViewController.m:
```diff
@@ -0,0 +1,172 @@
+    NSURL *url = [NSURL URLWithString:movie.imagePath];
+    if (self.isGrid) {
+        BoxOfficeCollectionCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:NSStringFromClass([BoxOfficeCollectionCell class])
+                                                                                      forIndexPath:indexPath];
```
Misaligned colons.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeTableCell.h:
```diff
@@ -0,0 +1,16 @@
+@interface BoxOfficeTableCell : UICollectionViewCell
+
+- (void)setThumbnailImageFromURL:(NSURL *)url;
+- (void)setName:(NSString *)name;
```
As with the other cell, I'd say it's preferable to have a method like `displayMovie:` that takes a model object.

vokal-isaac commented on MovieChecker/MovieChecker/BoxOfficeTableCell.m:
```diff
@@ -0,0 +1,37 @@
+                                                                NSError *error,
+                                                                SDImageCacheType cacheType,
+                                                                NSURL *imageURL) {
+        self.thumbnailImageView.image = image;
```
Probably safer as a weak reference to `self`...  and, actually, as much overlap as there is in your two cell classes, it would probably make sense to have a common base cell class to take care of the common code.

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.h:
```diff
@@ -0,0 +1,17 @@
+
+@interface MovieDetailViewController : UITableViewController
+
+@property (nonatomic, weak) Movie *movie;
```
I would make this a `strong` reference—it wouldn't make sense for the view controller to continue on without the movie model object.

vokal-isaac commented on MovieChecker/MovieChecker/MovieDetailViewController.m:
```diff
@@ -0,0 +1,34 @@
+@property (nonatomic, weak) IBOutlet UILabel *movieTitle;
+@property (nonatomic, weak) IBOutlet UILabel *rating;
+@property (nonatomic, weak) IBOutlet UILabel *synopsis;
+@property (nonatomic, weak) IBOutlet UILabel *year;
```
I would name these `...Label` so that the code reads more clearly—otherwise, they read like values rather than UI elements.

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.h:
```diff
@@ -0,0 +1,24 @@
+
+@end
+
+@interface MovieList : NSObject
```
`MovieList` reads like a model, but this is more like a controller... perhaps `MovieListController` or `MovieListLoader`?

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.h:
```diff
@@ -0,0 +1,24 @@
+
+@protocol MovieListDelegate <NSObject>
+
+- (void)synchMoviesWithArray:(NSArray *)movies;
```
Delegate methods should tell the receiver what the sender did (and, usually, should include a reference to the sender).  Perhaps something like `- (void)movieListController:(MovieListController *)movieListController didLoadMovies:(NSArray *)movies;`.

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.h:
```diff
@@ -0,0 +1,24 @@
+@property (nonatomic, strong) NSArray *movies;
+@property (nonatomic, weak) id <MovieListDelegate> delegate;
+
+- (id)initWithDelegate:(id)delegate;
```
It's a bit awkward that the action is caused by the initializer... it might be better for the action to be a separate method so that when this is consumed, it's clear than an action is undertaken and so that it'll be possible to repeat the action without having to throw away the object and get a new one.

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.m:
```diff
@@ -0,0 +1,81 @@
+                                              return;
+                                          }
+                                          NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
+                                          if (httpResp.statusCode != 200) {
```
Consider using [`ILGHttpConstants`](https://github.com/ilg/ILGHttpConstants).

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.m:
```diff
@@ -0,0 +1,81 @@
+                                              // TODO: Handle the bad status code!
+                                              return;
+                                          }
+                                          self.movies = [RottenTomatoesHelperMethods interpretBoxOfficeMoviesFromData:data
```
Probably should be a weak reference to `self`.

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.m:
```diff
@@ -0,0 +1,81 @@
+                                          self.movies = [RottenTomatoesHelperMethods interpretBoxOfficeMoviesFromData:data
+                                                                                                withkey:@"movies"];
+                                          [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+                                              [self.delegate synchMoviesWithArray:self.movies];
```
Probably should be weak references to `self`.

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.m:
```diff
@@ -0,0 +1,81 @@
+                                                   }
+                                                   movie.image = [[UIImage alloc] initWithData:data];
+                                               }];
+    [downloadImageTask resume];
```
It appears that the only purpose of the local variable `downloadImageTask` is to call `resume`—consider reformatting as:

``` objc
    [[[NSURLSession sharedSession]
      dataTaskWithURL:url
      completionHandler:^(NSData *data,
                          NSURLResponse *response,
                          NSError *error) {
          if (error) {
              // TODO: Handle the error!
              return;
          }
          NSHTTPURLResponse *httpResp = (NSHTTPURLResponse *)response;
          if (httpResp.statusCode != 200) {
              // TODO: Handle the bad status code!
              return;
          }
          movie.image = [[UIImage alloc] initWithData:data];
      }]
     resume];
```

vokal-isaac commented on MovieChecker/MovieChecker/MovieList.m:
```diff
@@ -0,0 +1,81 @@
+    self = [super init];
+    if (self) {
+        _delegate = delegate;
+        _movies = [NSMutableArray array];
```
The `movies` property isn't a mutable array; it also doesn't look like you'd make use of whatever value is saved here anyway.  Is there any need to set it at all?

vokal-isaac commented on MovieChecker/MovieChecker/RottenTomatoesHelperMethods.h:
```diff
@@ -0,0 +1,16 @@
+@interface RottenTomatoesHelperMethods : NSObject
+
++ (NSURL *)boxOfficeURL;
++ (NSArray *)interpretBoxOfficeMoviesFromData:(NSData *)data withkey:(NSString *)key;
```
Is there a reason for these to exist outside of `MovieList`?  If not, consider moving them inside that class.

vokal-isaac commented on MovieChecker/MovieCheckerTests/MovieCheckerTests.m:
You probably don't need this with `NSURLSession`.

vokal-isaac commented on MovieChecker/MovieCheckerTests/MovieCheckerTests.m:
Ditto.

vokal-isaac commented on MovieChecker/MovieCheckerTests/MovieCheckerTests.m:
It looks like this method tests `+ [RottenTomatoesHelperMethods boxOfficeURL]` and `+ [RottenTomatoesHelperMethods interpretBoxOfficeMoviesFromData:]`, but nothing from `MovieList`, which is the class that does the actual list-loading in your app.

vokal-isaac commented:
I'm going to merge this down so that we can start on a new pull request tomorrow.  Let's make sure that we talk about how to use branches in git to keep any one pull request more limited in scope while allowing you to work on other things on the side.
